### ISSUE TITLE
Consider running Fontspector checks also on sources before build

### ISSUE THREAD

**Issue by RickyDaMa** [googlefonts/googlesans-flex#945]
Currently, they're run afterwards, and only if the font builds, which seems illogical

It also means that if the font doesn't build, the sources aren't checked, which may have helped to diagnose the issue

Two approaches we could instead consider:

1. Run the source checks before building the font, and only build if the checks pass (no fails)
    - Should take roughly the same amount of time, maybe slightly longer
    - Might prevent the font from being built when it otherwise would have

2. Run the source checks alongside building the font
    - Will speed up the pipeline, as more work can be done at once
    - We may end up wasting time trying to build a font that we could have known would fail to build

**Comment by RickyDaMa** [comment]
Marianna chooses approach 1., but do not start work until after v2.003

**Comment by MariannaPaszkowska** [comment]
Let's also make sure to test the binaries, after we run checks on sources. 
Context: FB might have some checks that are only on binary.

**Comment by RickyDaMa** [comment]
Yes of course, the overwhelmingly majority of Fontbakery checks are on compiled fonts, zero intention of removing them!

**Comment by chrissimpkins** [comment]
@felipesanches @simoncozens are we doing this in the expressive catalog or Noto projects?  Do we have a significant number of source-based FB checks at this stage to warrant putting time into this?

**Comment by chrissimpkins** [comment]
@belluzj @MariannaPaszkowska do you have custom checks in mind?  If so, what are they and would they be more appropriate upstreamed into general FB?

**Comment by simoncozens** [comment]
I doubt that we have enough source checks to make it worthwhile.

**Comment by felipesanches** [comment]
> [@felipesanches](https://github.com/felipesanches) [@simoncozens](https://github.com/simoncozens) are we doing this in the expressive catalog or Noto projects? Do we have a significant number of source-based FB checks at this stage to warrant putting time into this?

The codebase is ready to support source-level checks, but there's probably only one example check (if I recall correctly) to showcase the capability. I think it would only make sense if we were willing to put the majority of the effort into actually implementing a batch of new source-level checks. For that, the workflow would be the same as usual: send a PR with a new implementation for me to review, or open an issue with a check proposal for us to discuss.

**Comment by felipesanches** [comment]
Please also consider the eventual challenge of porting such new checks to rust in the near future, as this is the current prospect for fontbakery (currently being prototyped as the fontspector project at https://github.com/simoncozens/fontspector/).

**Comment by RickyDaMa** [comment]
> Do you have custom checks in mind? If so, what are they and would they be more appropriate upstreamed into general FB?

@chrissimpkins you can see the current source checks GS Flex uses [here](https://github.com/googlefonts/googlesans-flex/blob/395910380a2280ecf3763be6c3cb14813d9bb106/qa/checks/sources.py). None of these make sense to upstream to Fontbakery from my previous review

The ones that could be upstreamed have been, although we don't yet take advantage of them in this repo. See #983 for the checks I've open-sourced

**Comment by MariannaPaszkowska** [comment]
from the team meeting: It's about reordering the checks, not removing ones or adding new ones.
